service/src/main/java/bio/terra/catalog/api/GlobalExceptionHandler.java

```diff
@@ -1,38 +1,16 @@
 package bio.terra.catalog.api;
 
 import bio.terra.catalog.model.ErrorReport;
-import bio.terra.common.exception.UnauthorizedException;
-import javax.annotation.Nonnull;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import bio.terra.common.exception.AbstractGlobalExceptionHandler;
+import java.util.List;
 import org.springframework.http.HttpStatus;
-import org.springframework.web.bind.annotation.ExceptionHandler;
-import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
 
 @RestControllerAdvice
-public class GlobalExceptionHandler {
-  private static final Logger logger = LoggerFactory.getLogger(GlobalExceptionHandler.class);
+public class GlobalExceptionHandler extends AbstractGlobalExceptionHandler<ErrorReport> {
 
-  @ExceptionHandler(UnauthorizedException.class)
-  @ResponseStatus(HttpStatus.UNAUTHORIZED)
-  public ErrorReport samAuthorizationException(UnauthorizedException e) {
-    return buildErrorReport(e);
-  }
-
-  @ExceptionHandler(Exception.class)
-  @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
-  public ErrorReport backstopHandler(Exception e) {
-    logger.error("Exception caught by backstop handler", e);
-    return buildErrorReport(e);
-  }
-
-  private ErrorReport buildErrorReport(@Nonnull Throwable e) {
-    StringBuilder combinedCauseString = new StringBuilder();
-    for (Throwable cause = e; cause != null; cause = cause.getCause()) {
-      combinedCauseString.append("cause: ").append(cause).append(", ");
-    }
-    logger.error("Global exception handler: " + combinedCauseString, e);
-    return new ErrorReport().message(e.getMessage());
+  @Override
+  public ErrorReport generateErrorReport(Throwable ex, HttpStatus statusCode, List<String> causes) {
```

Comment:
```suggestion
  public ErrorReport reportFromError(Throwable ex, HttpStatus statusCode, List<String> causes) {
```

Reply:
I like this idea but it's an `Override` of a tcl method so we don't have control over the name.

Reply:
Oh yeah whoops :(